tests/test_core.py

```diff
@@ -317,6 +317,18 @@ def test_filter_day(test_df, test_day):
         assert unique_time[0] == expected
 
 
+def test_filter_numpy_vals(test_df):
+    dates = test_df[test_df.time_col].unique()
+    if test_df.time_col == "year":
```

Comment:
I think this pattern would be a little cleaner as:

```
key = 'year' if test_df.time_col == "year" else 'time'
res_0 = test_df.filter(**{key: dates[0]})
res = test_df.filter(**{key: dates})
```

Reply:
That is neater, thanks